File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    n = len(sequence)
    s,end = 0, 0
    for i in range(len(sequence)):
        while s < k and end < n:
            s += sequence[end]
            end += 1
        if s == k:
            answer.append((i,end-1))
        s -= sequence[i]
            
    return sorted(answer,key=lambda x: x[1]-x[0])[0]

File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = {0,1000001};
        int end = 0;
        int total = 0;
        for (int i = 0; i < sequence.length; i++){
            while (total < k && end < sequence.length){
                total += sequence[end];
                end++;
            }
            if ( total == k && (end-i-1) < (answer[1]-answer[0]) ){
                answer[0] = i;
                answer[1] = end-1;
            }
            total -= sequence[i];
        };
        
        return answer;
    }
}